examples/fit-model-ceres.cpp

```diff
@@ -83,406 +262,159 @@ std::ostream& operator<<(std::ostream& out, const std::vector<T>& v)
-    {
-        landmarks = core::read_pts_landmarks(landmarksfile.string());
-    } catch (const std::runtime_error& e)
-    {
-        cout << "Error reading the landmarks: " << e.what() << endl;
-        return EXIT_FAILURE;
-    }
-    morphablemodel::MorphableModel morphable_model;
-    try
-    {
-        morphable_model = morphablemodel::load_model(modelfile.string());
-    } catch (const std::runtime_error& e)
-    {
-        cout << "Error loading the Morphable Model: " << e.what() << endl;
+    // Read all data from disk
+    ceres_example::FittingData fitting_data;
+    try {
+        fitting_data = ceres_example::read_fitting_data(cli_arguments.blendshapesfile.string(),
+                                                        cli_arguments.contourfile.string(),
+                                                        cli_arguments.imagefile.string(),
+                                                        cli_arguments.landmarksfile.string(),
+                                                        cli_arguments.mappingsfile.string(),
+                                                        cli_arguments.modelfile.string(),
+                                                        cli_arguments.outputfile.string());
+    } catch (const std::runtime_error& e) {
         return EXIT_FAILURE;
     }
 
-    // Note: Actually it's a required argument, so it'll never be empty.
-    core::LandmarkMapper landmark_mapper =
-        mappingsfile.empty() ? core::LandmarkMapper() : core::LandmarkMapper(mappingsfile.string());
-
-    std::vector<eos::morphablemodel::Blendshape> blendshapes =
-        eos::morphablemodel::load_blendshapes(blendshapesfile.string());
-
-    // Draw the loaded landmarks:
-    Mat outimg = image.clone();
-    for (const auto& lm : landmarks)
-    {
-        cv::rectangle(outimg, cv::Point2f(lm.coordinates[0] - 2.0f, lm.coordinates[1] - 2.0f),
-                      cv::Point2f(lm.coordinates[0] + 2.0f, lm.coordinates[1] + 2.0f), {255, 0, 0});
-    }
+    const auto& model_contour = fitting_data.model_contour;
+    const auto& ibug_contour = fitting_data.ibug_contour;
+    const auto& image = fitting_data.image;
+    const auto& morphable_model = fitting_data.morphable_model;
+    const auto& landmark_mapper = fitting_data.landmark_mapper;
+    const auto& blendshapes = fitting_data.blendshapes;
 
-    constexpr bool use_perspective = false;
+    // These will be the 2D image points and their corresponding 3D vertex id's used for the fitting
+    auto& landmarks = fitting_data.landmarks;
 
-    // These will be the 2D image points and their corresponding 3D vertex id's used for the fitting:
-    vector<Vector2f> image_points; // the 2D landmark points
-    vector<int> vertex_indices; // their corresponding vertex indices
+    auto landmarks_definitions = morphable_model.get_landmark_definitions();
+    auto* landmarks_definitions_ptr = landmarks_definitions.has_value() ? &landmarks_definitions.value() : nullptr;
```

Comment:
I don't think we should use a pointer here, it makes the code unnecessarily harder. If the value is optional, it would be good to use `eos::cpp17::optional`, which will typedef to `std::optional` on C++17 compilers.

Reply:
How do you think is it is a good idea to use `eos::cpp17::optional` in `landmark_mapper.get_indexed_landmarks` as a 2nd argument instead of a pointer? I decided to use a pointer here to be able to pass it in this function. Anyway, if I need to pass the pointer to the function I need to make it somehow (using `if` or ternary operator). I can use `if` and pass different arguments in case of empty `optional`, but it is more code :-) What do you think?

Reply:
Ok. Will change it.